internal/uvm/vsmb.go

```diff
@@ -32,31 +34,54 @@ func (uvm *UtilityVM) AddVSMB(ctx context.Context, hostPath string, guestRequest
 
 	uvm.m.Lock()
 	defer uvm.m.Unlock()
-	share, err := uvm.findVSMBShare(ctx, hostPath)
+
+	st, err := os.Stat(hostPath)
+	if err != nil {
+		return err
+	}
+	var file string
+	m := uvm.vsmbDirShares
+	if !st.IsDir() {
+		m = uvm.vsmbFileShares
+		file = hostPath
+		hostPath = filepath.Clean(filepath.Dir(hostPath))
+		options.RestrictFileAccess = true
+		options.SingleFileMapping = true
+	}
+	var requestType = requesttype.Update
+	share, err := uvm.findVSMBShare(ctx, m, hostPath)
 	if err == ErrNotAttached {
+		requestType = requesttype.Add
 		uvm.vsmbCounter++
 		shareName := "s" + strconv.FormatUint(uvm.vsmbCounter, 16)
 
+		share = &vsmbShare{
+			name:         shareName,
+			guestRequest: guestRequest,
+		}
+	}
+	allowedFiles := append(share.allowedFiles, file)
```

Comment:
Only `if options.RestrictFileAccess` ?

Reply:
We set both `file` and `options.RestrictFileAccess` in the same place above, so `file` won't ever be set unless `options.RestrictFileAccess` is as well.

Reply:
You're right, I'm adding an empty string each time here, that's crazy. :)